### PR TITLE
[Refactor](Variant) make many insterfaces exception safe

### PR DESCRIPTION
**Pull request by eldenmoon**
## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

File: be/src/vec/columns/column_object.cpp [hunk 1]
```diff
@@ -1552,9 +1552,11 @@ void ColumnObject::update_hash_with_value(size_t n, SipHash& hash) const {
...
     for_each_imutable_subcolumn([&](const auto& subcolumn) {
         if (n >= subcolumn.size()) {
-            LOG(FATAL) << n << " greater than column size " << subcolumn.size()
-                       << " sub_column_info:" << subcolumn.dump_structure()
-                       << " total lines of this column " << num_rows;
+            std::stringstream ss;
+            ss << n << " greater than column size " << subcolumn.size()
+               << " sub_column_info:" << subcolumn.dump_structure()
+               << " total lines of this column " << num_rows;
+            throw doris::Exception(ErrorCode::INTERNAL_ERROR, ss.str());
```
Comment (yiguolei) [inline be/src/vec/columns/column_object.cpp:1559]: exception 接口有格式化string的功能，比如 throw doris::Exception(ErrorCode::INTERNAL_ERROR, “afaf {}”, 123); 可以这么写

**Comment by eldenmoon** [comment]
run buildall

File: be/src/vec/columns/column_object.cpp (hunk 1 above)
Comment (eldenmoon) [reply be/src/vec/columns/column_object.cpp]: done

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by csun5285 — APPROVED** [review]
LGTM

**Review by airborne12 — COMMENTED** [review]
LGTM

**Pull request merged**